subsys/bluetooth/mesh/rpl.c

```diff
@@ -344,3 +348,39 @@ void bt_mesh_rpl_pending_store(uint16_t addr)
 		}
 	}
 }
+
+size_t bt_mesh_rpl_get(void (*cb)(const struct bt_mesh_rpl *rpl_entry))
+{
+	struct bt_mesh_rpl entry;
+	size_t count = 0;
+
+	if (!cb) {
+		return 0;
+	}
+
+	for (int i = 0; i < ARRAY_SIZE(replay_list); i++) {
+		if (replay_list[i].src) {
+			entry = replay_list[i];
+		} else {
+			break;
```

Comment:
This will break the for loop. But the entries in the replay_list is not necessary continues. I believe that if a entry is deleted, there can be blank "src" in the middle of the list.

Reply:
There is no ability to delete entries from the list the only possibility is to clean up еру RPL entirely. Entries are allocated one by one when it meets the first empty cell in the list.
I can add walking through the full list but it doesn't have sense.